src/test/java/teammates/sqlui/webapi/GetStudentActionTest.java

```diff
@@ -0,0 +1,288 @@
+        String[] params = {
+                Const.ParamsNames.COURSE_ID, stubCourse.getId(),
+                Const.ParamsNames.STUDENT_ID, stubStudent.getGoogleId(),
+        };
+        when(mockLogic.getStudentByGoogleId(stubCourse.getId(), stubStudent.getGoogleId())).thenReturn(null);
+        verifyEntityNotFound(params);
+    }
+
+    @Test
+    void testSpecificAccessControl_instructorWithPermission_canAccess() {
+        loginAsInstructor(stubInstructor.getGoogleId());
+        when(mockLogic.getCourse(stubCourse.getId())).thenReturn(stubCourse);
+        when(mockLogic.getStudentForEmail(stubCourse.getId(), stubStudent.getEmail())).thenReturn(stubStudent);
+        when(mockLogic.getInstructorByGoogleId(stubCourse.getId(), stubInstructor.getGoogleId()))
+                .thenReturn(stubInstructor);
+        String[] params = {
+                Const.ParamsNames.COURSE_ID, stubCourse.getId(),
+                Const.ParamsNames.STUDENT_EMAIL, stubStudent.getEmail(),
+        };
+        verifyCanAccess(params);
+
+        logoutUser();
+        verifyCannotAccess(params);
+    }
+
+    @Test
+    void testSpecificAccessControl_instructorWithoutPermission_cannotAccess() {
+        loginAsInstructor(stubInstructor.getGoogleId());
+        Instructor stubInstructorWithoutPermission = new Instructor(stubCourse, "name", "google.com",
+                false, "googleId",
+                InstructorPermissionRole.INSTRUCTOR_PERMISSION_ROLE_CUSTOM, new InstructorPrivileges());
+        when(mockLogic.getCourse(stubCourse.getId())).thenReturn(stubCourse);
+        when(mockLogic.getStudentForEmail(stubCourse.getId(), stubStudent.getEmail())).thenReturn(stubStudent);
+        when(mockLogic.getInstructorByGoogleId(stubCourse.getId(), stubInstructor.getGoogleId()))
+                .thenReturn(stubInstructorWithoutPermission);
+        String[] params = {
+                Const.ParamsNames.COURSE_ID, stubCourse.getId(),
+                Const.ParamsNames.STUDENT_EMAIL, stubStudent.getEmail(),
+        };
+        verifyCannotAccess(params);
+    }
+
+    @Test
+    void testSpecificAccessControl_studentPersonalAccess_canAccess() {
+        loginAsStudent(stubStudent.getGoogleId());
+        when(mockLogic.getCourse(stubCourse.getId())).thenReturn(stubCourse);
+        when(mockLogic.getStudentByGoogleId(stubCourse.getId(), stubStudent.getGoogleId())).thenReturn(stubStudent);
+
+        String[] params = {
+                Const.ParamsNames.COURSE_ID, stubCourse.getId(),
+        };
+        verifyCanAccess(params);
+
+        logoutUser();
+        verifyCannotAccess(params);
+    }
+
+    @Test
+    void testSpecificAccessControl_studentPublicAccess_cannotAccess() {
+        loginAsStudent(stubStudent.getGoogleId());
+        when(mockLogic.getCourse(stubCourse.getId())).thenReturn(stubCourse);
+        when(mockLogic.getStudentByGoogleId(stubCourse.getId(), stubStudent.getGoogleId())).thenReturn(stubStudent);
```

Comment:
If i understand this test case correctly this should be
```suggestion
        when(mockLogic.getStudentForEmail(stubCourse.getId(), "randomEmail")).thenReturn(null);;
```